plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/model/PostgreDialect.java

```diff
@@ -716,6 +737,14 @@ public void initDriverSettings(JDBCSession session, JDBCDataSource dataSource, J
         );
 
         addExtraKeywords(POSTGRE_EXTRA_KEYWORDS);
+
+        // Add Extra keywords from Redshift
+        if (((PostgreDataSource) dataSource).getServerType() instanceof PostgreServerRedshift) {
+            addExtraKeywords(REDSHIFT_EXTRA_KEYWORDS);
+            addKeywords(Arrays.asList(REDSHIFT_OTHER_TYPES_FUNCTION), DBPKeywordType.OTHER);
+            addExtraFunctions(REDSHIFT_FUNCTIONS_CONDITIONAL);
+        }
+
```

Comment:
You can move this snippet a bit lower, so you can use `serverExtension` field from `PostgreDialect` itself:

https://github.com/dbeaver/dbeaver/blob/b32d53917b49383a5987051c9c5492e016fd0023/plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/model/PostgreDialect.java#L781-L784

Reply:
Actually, it would be better to override `PostgreServerExtension#configureDialect` in `PostgreServerRedshift` and add keywords to the dialect there

Reply:
Good idea! Done and commits are squashed.